src/pixelmapFeature.js

```diff
@@ -0,0 +1,451 @@
+   * @returns {geo.pixelmap}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.mapColor = function (val) {
+    if (val === undefined) {
+      return m_this.style('mapColor');
+    } else if (val !== m_this.style('mapColor')) {
+      m_this.style('mapColor', val);
+      m_this.dataTime().modified();
+      m_this.modified();
+    }
+    return m_this;
+  };
+
+  /**
+   * If the specified coordinates are in the rendered quad, use the basis
+   * information from the quad to determine the pixelmap index value so that it
+   * can be included in the found results.
+   */
+  this.pointSearch = function (coordinate) {
+    if (m_quadFeature && m_info) {
+      var result = m_quadFeature.pointSearch(coordinate);
+      if (result.index.length === 1 && result.extra && result.extra[result.index[0]].basis) {
+        var basis = result.extra[result.index[0]].basis, x, y, idx;
+        x = Math.floor(basis.x * m_info.width);
+        y = Math.floor(basis.y * m_info.height);
+        if (x >= 0 && x < m_info.width &&
+            y >= 0 && y < m_info.height) {
+          idx = m_info.indices[y * m_info.width + x];
+          result = {
+            index: [idx],
+            found: [m_this.data()[idx]]
+          };
+          return result;
+        }
+      }
+    }
+    return {index: [], found: []};
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * Build
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._build = function () {
+    /* Set the build time at the start of the call.  A build can result in
+     * drawing a quad, which can trigger a full layer update, which in tern
+     * checks if this feature is built.  Setting the build time avoid calling
+     * this a second time. */
+    m_this.buildTime().modified();
+    if (!m_srcImage) {
+      var src = this.style.get('url')();
+      if (src instanceof Image && src.complete && src.naturalWidth && src.naturalHeight) {
+        /* we have an already laoded image, so we can just use it. */
+        m_srcImage = src;
+        this._computePixelmap();
+      } else if (src) {
+        var defer = new $.Deferred(), prev_onload, prev_onerror;
+        if (src instanceof Image) {
+          /* we have an unloaded image.  Hook to the load and error callbacks
+           * so that when it is loaded we can use it. */
+          m_srcImage = src;
+          prev_onload = src.onload;
+          prev_onerror = src.onerror;
+        } else {
+          /* we were given a url, so construct a new image */
+          m_srcImage = new Image();
+          // Only set the crossOrigin parameter if this is going across origins.
+          if (src.indexOf(':') >= 0 &&
+              src.indexOf('/') === src.indexOf(':') + 1) {
+            m_srcImage.crossOrigin = this.style.get('crossDomain')() || 'anonymous';
+          }
+        }
+        m_srcImage.onload = function () {
```

Comment:
The onload function is used on both a newly created Image and on an existing Image that isn't complete.  It is within the `else if` clause on line 180.